src/Microsoft.AspNet.Buffering/ResponseBufferingMiddleware.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Buffering
+{
+    public class ResponseBufferingMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ResponseBufferingMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            var originalBufferingFeature = httpContext.GetFeature<IHttpBufferingFeature>();
```

Comment:
If HasStarted, Invoke shouldn't even be called. Do we have any scenarios where middleware writes to the response and then calls next?